### DISCUSSION TITLE
allow passing pattern to sometimesBy and friends?

### DISCUSSION THREAD

**Discussion by felixroos** [tidalcycles/strudel#198] [Ideas]
this would be nice to randomly switch between patterns:

```js
"bd".sometimes("hh").s().fast(4).out()
```

can be done with

```js
Pattern.prototype._sometimesBy = function (x, other) {
  other = typeof other === 'function' ? other(this._undegradeBy(1 - x)) : reify(other)._undegradeBy(1 - x);
  return stack(this._degradeBy(x), other);
};
```

**Comment by yaxu** [comment]
It reads well but I wonder if it gets in the way of understanding what's going on / what's possible.

Something more explicit:
```js
"bd".sometimes(replace("hh")).s().fast(4).out()
```

In haskell `replace` would be `const`. `util.js` defines this as `constant` but it's not (yet) curried, and it's not the nicest name.

**Reply by felixroos** [reply]
hm yes it's actually much more readable that way

**Reply by yaxu** [reply]
Actually we already have `set` for this

https://strudel.tidalcycles.org?VGsjmHzmkMz0

**Reply by yaxu** [reply]
Somehow `squeeze`/`setSqueeze` isn't a top level function yet but this works well

```js
"bd".sometimes(x => x.squeeze("hh hh hh")).s().fast(4).out()
```

**Reply by yaxu** [reply]
```js
"bd@2 bd".sometimes(x => x.squeeze("hh hh hh")).s().out()
```
